c/driver/sqlite/statement_reader.c

```diff
@@ -217,25 +218,25 @@ static AdbcStatusCode ArrowTimestampToIsoString(int64_t value, enum ArrowTimeUni
   const time_t time = seconds;
 #endif
 
+  struct tm* tmPtr;
   struct tm broken_down_time;
 
-#if defined(_WIN32)
-  if (gmtime_s(&broken_down_time, &time) != 0) {
-    SetError(error,
-             "Could not convert timestamp %" PRId64 " with unit %d to broken down time",
-             value, unit);
-
-    return ADBC_STATUS_INVALID_ARGUMENT;
-  }
+#if HAVE_GMTIME_R
+  tmPtr = gmtime_r(&time, &broken_down_time);
 #else
-  if (gmtime_r(&time, &broken_down_time) != &broken_down_time) {
+  sqlite3_mutex_enter(sqlite3_mutex_alloc(SQLITE_MUTEX_STATIC_MAIN));
```

Comment:
Hmm, the SQLite docs state that we're not supposed to use this mutex

Reply:
Nice catch - I just replaced this with the mutex for the sqlite3 connection, which is used elsewhere. That may be a little overkill but I figured easier to implement than a separate static mutex

Not an expert in multithreaded C code so definitely open to feedback